src/assets.rs

```diff
@@ -273,12 +284,26 @@ impl HighlightingAssets {
         self.syntax_set
             .find_syntax_by_extension(file_name.to_str().unwrap_or_default())
             .or_else(|| {
+                let file_path = Path::new(file_name);
                 self.syntax_set.find_syntax_by_extension(
-                    Path::new(file_name)
+                    file_path
                         .extension()
                         .and_then(|x| x.to_str())
                         .unwrap_or_default(),
-                )
+                ).or_else(|| {
+                    let file_str = file_path.to_str().unwrap_or_default();
+                    for suffix in IGNORED_SUFFIXES.iter() {
+                        if file_str.ends_with(suffix) {
+                            return self.get_extension_syntax(
+                                OsStr::new(
+                                    file_str
+                                        .strip_suffix(suffix)
+                                        .unwrap_or_default()
+                                ));
+                        }
```

Comment:
Instead of using `.ends_with(suffix)`, `.strip_suffix(suffix)` and finally `.unwrap*`, I think you could simply call `strip_suffix` and decide based on the result. Something like:
```rust
if let Some(stripped_filename) = file_str.strip_suffix(suffix) {
    return self.get_extension_syntax(OsStr::new(stripped_filename));
}
```

Reply:
Good stuff, TIL, done.